src/hist/basehist.py

```diff
@@ -320,7 +325,15 @@ def plot1d(
 
         import hist.plot
 
-        return hist.plot.histplot(self, ax=ax, **_proc_kw_for_lw(kwargs))
+        if self.ndim == 1:
+            return hist.plot.histplot(self, ax=ax, **_proc_kw_for_lw(kwargs))
+        if overlay is not None:
+            cat_ax = self.axes[overlay]
+        else:
+            (cat_ax,) = (ax for ax in self.axes if ax.traits.discrete)
+        cats = cat_ax if cat_ax.traits.discrete else np.arange(len(cat_ax.centers))
+        d1hists = [self[{cat_ax.name: cat}] for cat in cats]
```

Comment:
```suggestion
        if overlay is None:
            (overlay,) = (i for i, ax in enumerate(self.axes) if ax.traits.discrete)
        assert overlay is not None
        cat_ax = self.axes[overlay]
        cats = cat_ax if cat_ax.traits.discrete else np.arange(len(cat_ax.centers))
        d1hists = [self[{overlay: cat}] for cat in cats]
```

Maybe something like this would support unnamed axes? The assert is only needed if MyPy needs it (I think it does).